### PR TITLE
Update linux flow on download-thanks

### PR DESCRIPTION
**Pull request by nathan-barrett**
## One-line summary
Updates the linux download flow and adds a link to the instructions for Debian based distributions on SUMO

## Significant changes and points to review
Update the body class to `linux` and `linux x64` to test these pages for both architecture 
 / - or - \
in `site.js` change the return value of `getPlatform` to `linux`  and the value of `getArchSize` to `64` to help expedite the testing process.

## Issue / Bugzilla link
#14012 
#12966 

## Testing
http://localhost:8000/firefox/download/thanks/
http://localhost:8000/firefox/all#product-desktop-release
http://localhost:8000/firefox/linux/

### CODE REVIEW

File: bedrock/firefox/templates/firefox/includes/download-button.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/includes/download-button.html:65]: Any specific reason why this is being removed?

File: bedrock/firefox/templates/firefox/new/desktop/thanks.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/new/desktop/thanks.html:83]: Might it be worth using a different title here for Linux? The title here kind of implies that the download has already happened. Maybe something like "Almost there! Select a download option below."?

image

File: l10n/en/brands.ftl
Comment (alexgibson) [inline l10n/en/brands.ftl:158]: I think we can remove this as we no longer use brand name variables for third parties

File: l10n/en/firefox/new/desktop.ftl
Comment (alexgibson) [inline l10n/en/firefox/new/desktop.ftl:217]: There should be no need to repeat these strings as `download_button.ftl` is a global string file that all pages can access.

You could probably add all the strings (including the "using debian" one) to the global download button file?

File: media/css/firefox/channel.scss
Comment (alexgibson) [inline media/css/firefox/channel.scss:50]: Might it be worth trying to keep download button specific selectors in `_download-button.scss`, or would it cause issues elsewhere?

**Review by alexgibson — COMMENTED** [review]
Left a few comments, but overall approach looks good to me 👍

File: bedrock/firefox/templates/firefox/includes/download-button.html
Comment (nathan-barrett) [reply bedrock/firefox/templates/firefox/includes/download-button.html:65]: Oop, that was a mistake. good 👁️

File: bedrock/firefox/templates/firefox/all-unified.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/all-unified.html:150]: Nit: remove `/en-US/` from SUMO link

File: bedrock/firefox/templates/firefox/includes/macros.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/includes/macros.html:72]: Nit: remove `/en-US/` from SUMO link

File: bedrock/firefox/templates/firefox/new/desktop/thanks.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/new/desktop/thanks.html:90]: Nit: remove `/en-US/` from SUMO link

File: l10n/en/download_button.ftl
Comment (alexgibson) [inline l10n/en/download_button.ftl:67]: Nit: remove third party brand name `{ -brand-name-ubuntu }`
```suggestion
download-button-using-debian = Using Debian, Ubuntu or any Debian-based distribution?<br>You can set up our <a { $attrs }>APT repository instead</a>.
```

File: l10n/en/download_button.ftl
Comment (alexgibson) [inline l10n/en/download_button.ftl:66]:
```suggestion
# Note: Debian and Ubuntu shouldn't be translated
```

File: l10n/en/download_button.ftl
Comment (alexgibson) [inline l10n/en/download_button.ftl:65]: Nit: remove `/en-US/` from SUMO link

File: l10n/en/firefox/new/platform.ftl
Comment (alexgibson) [inline l10n/en/firefox/new/platform.ftl:29]: Question: are these strings needed, seeing as they already exist in `download_button.ftl`?

File: media/css/firefox/all/all-unified.scss
Comment (alexgibson) [inline media/css/firefox/all/all-unified.scss:255]: Nit: remove blank lines

File: media/js/firefox/all/all-downloads-unified.js
Comment (alexgibson) [inline media/js/firefox/all/all-downloads-unified.js:446]: Question: Is this fixing something? The attributes should already get set in `setDownloadLink()` as far as I can see?

File: media/css/firefox/new/common/thanks-messaging.scss
Comment (alexgibson) [inline media/css/firefox/new/common/thanks-messaging.scss:37]: Issue: the buttons could use some slight alignment / spacing tweaks on /thanks? 

image

File: media/css/firefox/new/common/thanks-messaging.scss
Comment (alexgibson) [inline media/css/firefox/new/common/thanks-messaging.scss:94]: Issue: we should remove this `linux.arm` class altogether, since it now results in double messaging. 

We should probably remove the `arm` specific strings from the `/thanks/` page. And if we wanted to go a step further, we could remove the `arm` class detection from `site.js` too?

image

File: media/css/firefox/new/common/thanks-messaging.scss
Comment (alexgibson) [inline media/css/firefox/new/common/thanks-messaging.scss:85]: Issue: we need to remember to update the `basic.html` template, as right now I get a blank page?

http://localhost:8000/en-US/firefox/download/thanks/?xv=basic

image

File: media/css/firefox/channel.scss
Comment (alexgibson) [reply media/css/firefox/channel.scss:50]: I see this issue on the `/developer/` page too - and do we need to add the debian message there as well?

http://localhost:8000/en-US/firefox/developer/

image

File: bedrock/firefox/templates/firefox/all-unified.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/all-unified.html:149]: Spotted an issue: I don't see this string on initial page load if the page detects `linux` as my platform. It only appears after I change platform using the select box?

File: media/css/firefox/new/basic/platform.scss
Comment (alexgibson) [inline media/css/firefox/new/basic/platform.scss:13]: Issue: at smaller breakpoints the download buttons on this page are missing some padding

image

**Review by alexgibson — CHANGES_REQUESTED** [review]
I took a good pass and spotted a few visual bugs/issues here and there. The biggest thing we're missing I think is the new messaging on the `basic.html` thanks page?

Aside from that, I think it's mostly just nits and small button tweaks here and there. You've done a great job piecing together all the loosely cobbled parts here!

File: media/js/firefox/all/all-downloads-unified.js
Comment (nathan-barrett) [reply media/js/firefox/all/all-downloads-unified.js:446]: I added that to the form to use as a selector to show/hide the debian text that shows up at the bottom of the page near the download link. Let me mess around and maybe we can avoid doing this.

**Comment by alexgibson** [comment]
The developer edition page still has some button spacing issues on small view ports. It's also missing the SUMO link instructions?

image

**Comment by alexgibson** [comment]
The `/firefox/linux/` page still has some button spacing issues on mobile. Should we also perhaps display:none the line break on small screens for the SUMO instructions?

image

**Comment by alexgibson** [comment]
The `/firefox/developer/` page is missing the SUMO link. IS that intentional?

image

File: media/js/firefox/all/all-downloads-unified.js
Comment (alexgibson) [reply media/js/firefox/all/all-downloads-unified.js]: If this needs to stay, can we add a code comment to try and explain why it's here and what it's being used for?

File: bedrock/firefox/templates/firefox/new/desktop/thanks.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/new/desktop/thanks.html:88]: Issue: same as above, the has should come last after the query params

File: l10n/en/firefox/new/platform.ftl
Comment (alexgibson) [inline l10n/en/firefox/new/platform.ftl:19]: Nit: remove `/en-US/` from URL

File: media/css/firefox/all/all-unified.scss
Comment (alexgibson) [inline media/css/firefox/all/all-unified.scss:253]: Issue: this should use a `bidi()` mixing for RTL languages

File: media/css/firefox/new/common/thanks-messaging.scss
Comment (alexgibson) [inline media/css/firefox/new/common/thanks-messaging.scss:23]: Question: can we remove `show-linux-arm` if it's no longer used?

File: tests/unit/spec/firefox/new/common/thanks.js
Comment (alexgibson) [inline tests/unit/spec/firefox/new/common/thanks.js:122]: Question: can we remove this test?

File: media/js/firefox/all/all-downloads-unified.js
Comment (alexgibson) [inline media/js/firefox/all/all-downloads-unified.js:175]: Suggestion: can we add a code comment that explains what this is being used for?

**Review by alexgibson — CHANGES_REQUESTED** [review]
Spotted a few more small issues, but all the updates so far are looking good!

I notice on a few pages, such as http://localhost:8000/en-US/firefox/browsers/compare/, that the CTA on Linux is no longer very clear (since it doesn't mention Firefox directly)

image

We can always punt pages like this to a separate follow up issue if time is running short, but changing them to use `{{ download_firefox_thanks() }}` instead of `{{ download_firefox() }}` might be a simple solution.

Thoughts?

**Comment by alexgibson** [comment]
Oh also: let's make sure we push this to `run-integration-tests` before merging?

**Comment by nathan-barrett** [comment]
I _think_ I got to all of the issues you mentioned, @alexgibson  - and just pushed to `run-integration-tests` and will update when I have the results

**Comment by nathan-barrett** [comment]
> Spotted a few more small issues, but all the updates so far are looking good!
> 
> I notice on a few pages, such as http://localhost:8000/en-US/firefox/browsers/compare/, that the CTA on Linux is no longer very clear (since it doesn't mention Firefox directly)
> 
> image
> 
> We can always punt pages like this to a separate follow up issue if time is running short, but changing them to use `{{ download_firefox_thanks() }}` instead of `{{ download_firefox() }}` might be a simple solution.
> 
> Thoughts?

I went the route of having all of the buttons for linux to include `Download Lunix..`

**Review by alexgibson — APPROVED** [review]
Aside from the [one failing test](https://github.com/mozilla/bedrock/actions/runs/7614212907/job/20735973919) (which we discussed on Slack) I think everything looks good here. Nice work! r+

File: bedrock/firefox/templates/firefox/new/basic/thanks.html
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/new/basic/thanks.html:44]: Not a blocker, but might it be worth adding a new string here to match the `thanks.html` template? i.e. "Almost there! Select a download option below"

**Comment by nathan-barrett** [comment]
Successful integration tests: https://github.com/mozilla/bedrock/actions/runs/7616989923

File: l10n/en/firefox/new/platform.ftl
Comment (alexgibson) [inline l10n/en/firefox/new/platform.ftl:19]: Nit: could we restore the group comment here and the line space below?

**Review by alexgibson — APPROVED** [review]
r+wc - nice work on fixing that test!

I'll leave this to you to merge, since I'm not sure if you're waiting on any specific timing for the SUMO article or anything.

This was not a bit of a convoluted issue to tackle and you did a great job since it requires touching a lot of places. Nice work!

**Pull request merged**
